src/main/java/com/google/sps/utility/GeocodingResultUtility.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import com.google.maps.model.AddressType;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import java.util.Optional;
+
+/** Utility class to extract data from GeocodingResult objects. */
+public class GeocodingResultUtility {
+  /**
+   * Parses for coordinates in a resulting call to the Geocoding API.
+   *
+   * @param result A GeocodingResult returned from the Geocoding API.
+   * @return A LatLng representing coordinates.
+   */
+  public static LatLng getCoordinates(GeocodingResult result) {
+    return result.geometry.location;
+  }
+
+  /**
+   * Converts an AddressType to a PlaceType for the purpose of calling the Places API. PlaceType is
+   * a subset of AddressType and hence, not all AddressTypes are supported by the Places API.
+   *
+   * @param addressType An AddressType to convert to a PlaceType
+   * @return A PlaceType corresponding to an AddressType if available, null if not available.
+   */
+  public static Optional<PlaceType> convertAddressTypeToPlaceType(AddressType addressType) {
+    for (PlaceType placeType : PlaceType.values()) {
+      if (placeType.name().equals(addressType.name())) {
+        return Optional.ofNullable(placeType);
+      }
+    }
+    return Optional.empty();
```

Comment:
There is a more succinct way to write this using streams:

```java
placeType.values().stream()
  .filter(t -> t.name().equals(addressType.name()))
  .findFirst();
```
In particular, the stream method `findFirst` returns an Optional.empty() if the stream is empty.

There is also `findAny`, which is similar to `findFirst`, and may be more appropriate here semantically. The place type values aren't really ordered in a way that you care about.

Caveat: I am not sure if `placeType.values()` is a Collection or an Array, and whether or not Arrays support stream.

Reply:
Oh wow this is nice, I've tried it and I just needed to cast `PlaceType[]` to `List<PlaceType>`. Thanks!

Reply:
`RESOLVED`